docs/api.md

```diff
@@ -0,0 +1,531 @@
+## ArduinoCloud Class (Base)
+
+### `ArduinoCloud (Class)`
+
+#### Description
+
+`ArduinoIoTCloud` is a class for interfacing with the Arduino IoT Cloud. It provides an interface for managing IoT Cloud connectivity, updating data, and handling events.
+
+Depending on what type of connection is used (TCP/LPWAN), either the `ArduinoIoTCloudTCP` or `ArduinoIoTCloudLPWAN` base classes are initialized. This specification is done in the Arduino IoT Cloud's "Thing" interface, and is reflected in the automatically generated `thingProperties.h` file. 
+
+The above classes are inherited from the `ArduinoIoTCloudClass` base class, with methods used depending on what connection is used.
+
+#### Syntax
+
+```
+ArduinoCloud.method()
+```
+
+### `push()`
+
+#### Description
+
+Pushes data to the IoT Cloud.
+
+#### Syntax
+
+```
+ArduinoCloud.push()
+```
+
+#### Parameters
+
+None.
+
+#### Returns
+
+Nothing.
+
+### `setTimestamp()`
+
+Sets a timestamp for a cloud variable.
+
+#### Syntax
+
+```
+ArduinoCloud.setTimestamp(cloudVariable, timestamp)
+```
+
+#### Parameters
+
+- `String` - property name
+- `unsigned long const` - timestamp
+
+### `setThingId()`
+
+Sets the Thing ID.
+
+#### Syntax
+
+```
+ArduinoCloud.setThingId(thingId)
+```
+
+#### Parameters
+- `String` - your Thing's ID. Obtainable in the IoT Cloud interface. 
+
+### `String & getThingId()`
+
+Gets the Thing ID.
+
+#### Syntax
+
+#### Parameters
+
+### `setDeviceId()`
+
+Sets the Device ID.
+
+#### Syntax
+
+```
+ArduinoCloud.setDeviceId(deviceId)
+```
+
+#### Parameters
+
+- `String` - Device ID.
+
+### `getDeviceId()`
+
+Gets the Device ID.
+
+#### Syntax
+
+```
+ArduinoCloud.getDeviceId()
+```
+
+#### Parameters
+
+None
+
+#### Returns
+
+- `String` - Device ID.
+
+### `setThingIdOutdatedFlag()`
+
+Sets a flag indicating that the Thing ID is outdated.
+
+#### Syntax
+
+```
+ArduinoCloud.setThingIdOutdatedFlag()
+```
+
+#### Parameters
+None.
+
+#### Returns
+Nothing.
+
+### `clrThingIdOutdatedFlag()`
+
+Clears the flag indicating that the Thing ID is outdated.
+
+#### Syntax
+
+```
+ArduinoCloud.clrThingIdOutdatedFlag()
+```
+
+#### Parameters
+None.
+
+#### Returns
+Nothing.
+
+### `getThingIdOutdatedFlag()`
+
+Gets the flag indicating whether the Thing ID is outdated.
+
+#### Syntax
+
+```
+ArduinoCloud.getThingOutdatedFlag()
+```
+
+#### Parameters
+None.
+
+#### Returns
+- Boolean (true/false)
+
+### `deviceNotAttached()`
+
+Checks if the device is not attached.
+
+#### Syntax
+
+```
+ArduinoCloud.deviceNotAttached()
+```
+
+#### Parameters
+None.
+
+#### Returns
+
+- Boolean (true/false)
```

Comment:
I would not add this functions to the docs since are not intented to be used by the end user and we are planning to remove them from public API

Reply:
aha thanks for the clarification, it has been removed.